components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java

```diff
@@ -284,53 +293,94 @@ public String getUsernameByClaims(Map<String, String> claims, String tenantDomai
                     null);
         }
         int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
-        String[] resultedUserList = ArrayUtils.EMPTY_STRING_ARRAY;
-        for (String key : claims.keySet()) {
-            String value = claims.get(key);
-            if (StringUtils.isNotEmpty(key) && StringUtils.isNotEmpty(value)) {
-                if (log.isDebugEnabled()) {
-                    String message = String.format("Searching users by claim : %s for value : %s", key, value);
-                    log.debug(message);
+        try {
+            String[] userStoreDomainNames = getDomainNames(tenantId);
+            AbstractUserStoreManager carbonUM = (AbstractUserStoreManager) getUserStoreManager(tenantId);
+            RealmService realmService = IdentityRecoveryServiceDataHolder.getInstance().getRealmService();
+            ClaimManager claimManager = (ClaimManager) realmService.getTenantUserRealm(IdentityTenantUtil.
```

Comment:
we have the tenantId by now. We can use it here without resolving it from domain right?

Reply:
To get the claim manager , we have to pass tenant id to this getTenantUserRealm.

Reply:
Resolved by https://github.com/wso2-extensions/identity-governance/pull/623/commits/00a46fd2ecefd9f65eeebbacfa8e641f455a6a60